### ISSUE TITLE
Allow more flexible replacement

### ISSUE THREAD

**Issue by timkelty** [derek-duncan/postcss-responsify#2]
Hi - first wanted to say thanks for the plugin, it was just what I was looking for!

I have a pending PR, but wanted to float this by you first:

I need to support replacement more flexible than a mere _prefix_. For example, [SUITCSS](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md#utilities) has a naming convention like so:
- u-sm-utilityName, u-md-utilityName
- sm-ComponentName, md-ComponentName

So, I thought it would be nice if `postcss-responsify` could optionally accept a "replacement/transformer" option callback, that passed the selector and the prefix, and returned the transformed selector (this would be called within [`createPrefixedRule`](https://github.com/derek-duncan/postcss-responsify/blob/master/index.js#L89).

Might look something like:
```
const postcss = require('postcss');
const responsify = require('postcss-responsify');

const breakpoints = [
  { prefix: 'sm-', mediaQuery: '(min-width: 40em)' },
  { prefix: 'md-', mediaQuery: '(min-width: 52em)' },
];
const transformer = (selector, prefix) => selector.replace(/^\.(u-)?/, `.$1${prefix}`);
const responsifyOptions = {
  breakpoints,
  transformer
};
postcss().use(responsify(responsifyOptions));
```

This would also make it possible for users who wanted to do other naming schemes, such as a suffix, or anything more complicated.

In that light, it might make sense to rename `options.breakpoint.prefix` to something more generic, or even make it an object `prefix` is just the property.

Furthermore, we could even bundle a few of these common replacement strategies for an easier api:
```
postcss().use(responsify({
  breakpoints,
  transformer: 'suitcss' // could be `prefix` (default), `suffix`
}));

```
If `options.transformer` was a function we'd call it, but if string, we'd look for a default.

I'll work on a PR - but please let me know what you think, or if you have any other ideas.

**Comment by derek-duncan** [comment]
@timkelty Sure thing, I'll happily review a PR 👍  I like `transformer` as an option – that should cover all the use-cases. I agree that `breakpoints` should probably be an object instead of an array. But given that this package isn't very popular (see https://www.npmjs.com/package/postcss-responsify), I'd like to keep from pushing breaking changes. If you feel differently, we could cut a 2.0 release for these changes.

Thanks for contributing! 👍  Looking forward to that PR

**Comment by timkelty** [comment]
@derek-duncan Yep, might be easier to just call it 2.0 and not worry about breaking changes. If more people start using it we can always backport the transformer bit, but I'm guessing it wont be necessary.
